src/pages/permissions/actions/DeleteGroupModal.tsx

```diff
@@ -0,0 +1,73 @@
+import { ConfirmationModal, Input } from "@canonical/react-components";
+import { ChangeEvent, FC } from "react";
+import { LxdGroup } from "types/permissions";
+
+interface Props {
+  groups: LxdGroup[];
+  onConfirm: () => void;
+  disable?: boolean;
+  loading?: boolean;
+  onConfirmInputChange: (val: string) => void;
+  confirmInputValue: string;
+  confirmText: string;
+  close: () => void;
+}
+
+const DeleteGroupModal: FC<Props> = ({
+  groups,
+  onConfirm,
+  disable,
+  loading,
+  onConfirmInputChange,
+  confirmInputValue,
+  confirmText,
+  close,
+}) => {
+  const hasOneGroup = groups.length === 1;
+  const deleteGroupText = hasOneGroup
+    ? groups[0].name
+    : `${groups.length} groups`;
+  const handleConfirmInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    onConfirmInputChange(e.target.value);
+  };
+
+  return (
+    <ConfirmationModal
+      confirmButtonLabel={`Permanently delete ${deleteGroupText}`}
```

Comment:
Hmm I think there are two options here:
1. We can change the copy for the button to be something like "Permanently delete this group" in the case of a single group being deleted. There is a reference to the group name in the body of the modal anyway.
2. We truncate the text on the button and add a tooltip. This will require upstreaming to `ConfirmationModal` again though.

I'd prefer option 1, but I have noted this as a design question for @piperdeck as well.

Reply:
For now I have implemented option 1.

Reply:
Changing the copy for the button to always display number of groups deleted, even if it's 1 group.